libs/dataset-explorer/src/lib/ChartView/DataAnalysisView/SidePanel.tsx

```diff
@@ -62,7 +56,7 @@ export class SidePanel extends React.Component<
       text: localization.Interpret.DatasetExplorer.aggregatePlots
     },
     {
-      key: ifEnableLargeData(this.props.dataset)
+      key: ifEnableLargeData(this.context.dataset)
```

Comment:
won't the context only get populated at a later point? As in, this will run before `componentDidMount` so `this.context` will be undefined.

Reply:
yes, this is very weird behavior. When SidePanel is initialized from Other places the values are propagated into it via this.context. Not sure why context is not initialized at this point.

Reply:
That's just how the context works. If you want to pass something in you need to use props